### PR TITLE
fix(theme): improve color toggle when using dark navbar

### LINKED ISSUE #8587: Theme toggle :hover contrast issue while navbar style: 'dark'
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

while hovering on the theme toggle it still shows light background with light foreground color. this is a design flaw and leads to poor accessibility of the application. 
fixing this will improve the viewability of the button by more users and improve the accessibility of the application.

check the expected and current behaviors of the button below:

### Reproducible demo

_No response_

### Step 1
set the navbar style to 'dark' in `docusaurus.config.js` file
```
navbar: {
        title: "My Site",
        style: "dark",
```

### Step 2
while in light mode hover on the theme toggle

### Expected behavior

docusaurauscorrect

### Actual behavior

docusauraus

### Your environment

- Docusaurus version used: latest
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Chorme 89
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): Windows

### Self-service

- [X] I'd be willing to fix this bug myself.

### PR DESCRIPTION
**Pull request by dewanshDT**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

These changes are made to the issue #8587 this issue was leading to contrast Issue in color toggle button in the navbar. To reproduce this bug learn more in this issue #8587

### Test links

Deploy preview: https://deploy-preview-_____--docusaurus-2.netlify.app/

## Related issues/PRs
closes #8587

### CODE REVIEW

File: packages/docusaurus-theme-classic/src/theme/ColorModeToggle/styles.module.css [hunk 1]
```diff
@@ -25,6 +25,10 @@
   background: var(--ifm-color-emphasis-200);
 }
 
+.navbar--dark .toggleButton:hover {
+  background: var(--ifm-color-gray-800);
+}
```
Comment (slorber) [inline packages/docusaurus-theme-classic/src/theme/ColorModeToggle/styles.module.css:30]: this CSS selector does not work for me:

CleanShot 2023-02-02 at 11 02 33@2x


Please temporarily add a `style: 'dark'` to our site config so that we can see on the preview it works fine. We'll revert the change once it works before merging.

**Review by slorber — CHANGES_REQUESTED** [review]

File: packages/docusaurus-theme-classic/src/theme/ColorModeToggle/styles.module.css (hunk 1 above)
Comment (dewanshDT) [reply packages/docusaurus-theme-classic/src/theme/ColorModeToggle/styles.module.css:30]: hey I added `style: 'dark'` to config file

Comment (dewanshDT) [reply packages/docusaurus-theme-classic/src/theme/ColorModeToggle/styles.module.css:30]: navbardark
hey is there instead of `navbar--dark` it is `navbar--dark_Tx1g` is there any other file where I should put this style or is thre any other way to solve this

Comment (dewanshDT) [reply packages/docusaurus-theme-classic/src/theme/ColorModeToggle/styles.module.css:30]: I'm a bit new to this would really appreciate your help on this @slorber

Comment (dewanshDT) [reply packages/docusaurus-theme-classic/src/theme/ColorModeToggle/styles.module.css]: fixed it finally

**Comment by dewanshDT** [comment]
finally it is fixed 🎉, check the deploy preview @slorber

**Comment by dewanshDT** [comment]
sorry for too many commits :(

**Comment by dewanshDT** [comment]
okay

**Comment by dewanshDT** [comment]
@Josh-Cena sorry but that doesn't solve the issue #8587

**Comment by dewanshDT** [comment]
well adding `data-theme` to navbar doesn't solve this issue so instead of adding css I added `data-style` toggle and and checked for it this solves the problem and now it works

**Comment by dewanshDT** [comment]
Hey @Josh-Cena @slorber I've fixed the issue, are there any further changes you want me to make

**Review by slorber — APPROVED** [review]
Thanks, LGTM 👍 

Refactored a bit because things related to navbar should be in NavbarColorModeToggle instead of the more generic ColorModeToggle
